### PR TITLE
Ship API v3

### PR DESCRIPTION
**Pull request by humitos**
* Remove message about contacting support to get early API v3 access
* Remove all mentions to API v3 beta in the docs
* Mark API v2 as deprecated
* Add a note about where to get the access Token
* ~~Creates a Token each time a new User is created~~
* Allow the user to Generate/Revoke the Token under their profile page
* Standardize documentation (show only relevant fields when describing endpoints)
* Remove half-baked docstrings from Browsable API
* Remove not useful filters (Project, Version)
* Leave documentation for relevant fields only

Screenshot_2019-09-11_23-42-41

Screenshot_2019-09-11_23-42-52

~~Before merging this PR (or deploy it), we need to run a Python command to create a Token per User.~~

### CODE REVIEW

**Comment by davidfischer** [comment]
> Before merging this PR (or deploy it), we need to run a Python command to create a Token per User.

I gave this PR a short try and there was no way for users to create or revoke access tokens. That was fine for beta but I think for general use users need to be able to create and revoke them. If users can create their own access tokens, we don't need to create them for users and we can probably remove the signal too.

**Comment by humitos** [comment]
@davidfischer makes sense. I will make the changes to allow Create/Delete from the profile settings. Thanks!

**Comment by humitos** [comment]
@davidfischer I updated the PR with the ability to Generate/Revoke API Tokens under the profile page. Currently, it allows the user to have only one Token max. Let me know if you think this is the path to go forward.

Eventually, when we implement scope-based tokens, we will allow the user to manage more than one token at the same time.

**Comment by humitos** [comment]
Before shipping it, I'd like to do a second pass review to all the code (we may want to remove some non-useful filters, for example) and documentation (maybe a refactor or cleaning).

**Comment by humitos** [comment]
Removing "Status: blocked" label and calling for a last review of @readthedocs/core team before merging. I'd like to have this deployed in the next deploy hopefully.

**Comment by stsewd** [comment]
@humitos we should resolve what to do around privacy levels first

**Comment by stsewd** [comment]
The docs for redirects don't mention what kind of redirects can be created, and also, not all redirects require a `from` or `to` urls

**Comment by humitos** [comment]
Blocked on https://github.com/readthedocs/readthedocs.org/pull/6275 for now. We should merge that one first.

File: docs/api/v3.rst [hunk 1]
```diff
@@ -552,34 +523,6 @@ This allows for documentation projects to share a search index and a namespace o
 but still be maintained independently.
 See :doc:`/subprojects` for more information.
 
-
-Subproject details
-++++++++++++++++++
```
Comment (stsewd) [inline docs/api/v3.rst]: Looks like this shouldn't be deleted

File: docs/api/v3.rst [hunk 2]
```diff
@@ -791,14 +727,21 @@ Redirect create
...
         }
 
+    .. note::
+
+       ``type`` can be one of ``prefix``, ``page``, ``exact``, ``sphinx_html`` and ``sphinx_htmldir``.
+
+       Depending on the ``type`` of the redirect, some fields may not be needed:
+
+       * ``prefix`` type does not require ``to_url``.
+       * ``page`` and ``exact`` types require ``from_url`` and ``to_url``
```
Comment (stsewd) [inline docs/api/v3.rst]:
```suggestion
       * ``page`` and ``exact`` types require ``from_url`` and ``to_url``.
```

File: docs/api/v3.rst [hunk 3]
```diff
@@ -791,14 +727,21 @@ Redirect create
...
+       * ``sphinx_html`` and ``sphinx_htmldir`` types do not require ``from_url`` and ``to_url``
```
Comment (stsewd) [inline docs/api/v3.rst]:
```suggestion
       * ``sphinx_html`` and ``sphinx_htmldir`` types do not require ``from_url`` and ``to_url``.
```

File: docs/api/v3.rst (hunk 1 above)
Comment (humitos) [reply docs/api/v3.rst]: Hrm... I probably made a mistake when solving merge conflicts. Good catch!

**Comment by humitos** [comment]
> Some examples from the docs are still showing `privacy_level` in the response.

Can you point me to them? I'm not finding anyone.

**Comment by stsewd** [comment]
> Can you point me to them? I'm not finding anyone.

Can't see them anymore, guess merging master was missing?

**Comment by stsewd** [comment]
Looks like the failing tests are not related to the esling thing, btw

**Pull request merged**